securedrop/journalist_app/sessions.py

```diff
@@ -0,0 +1,278 @@
+            expires = self.lifetime
+        else:
+            if expires < (30 * 60) and session['renew_count'] > 0:
+                session['renew_count'] -= 1
+                expires += self.lifetime
+                session.modified = True
+        conditional_cookie_kwargs = {}
+        httponly = self.get_cookie_httponly(app)
+        secure = self.get_cookie_secure(app)
+        if self.has_same_site_capability:
+            conditional_cookie_kwargs["samesite"] = self.get_cookie_samesite(app)
+        val = self.serializer.dumps(dict(session))
+        if session.to_regenerate:
+            self.redis.delete(self.key_prefix + session.sid)
+            session.sid = self._generate_sid()
+            session.token = self._get_signer(app).dumps(session.sid)  # type: ignore
+        if session.new or session.to_regenerate:
+            self.redis.setex(name=self.key_prefix + session.sid, value=val,
+                             time=expires)
+        elif session.modified:
+            # To prevent race conditions where session is delete by an admin in the middle of a req
+            # accept to save the session object if and only if alrady exists using the xx flag
+            self.redis.set(name=self.key_prefix + session.sid, value=val,
+                           ex=expires, xx=True)
+        if not session.is_api and (session.new or session.to_regenerate):
+            response.headers.add('Vary', 'Cookie')
+            response.set_cookie(app.session_cookie_name, session.token,
+                                httponly=httponly, domain=domain, path=path,
+                                secure=secure, **conditional_cookie_kwargs)  # type: ignore
+
+
+class Session(object):
+
+    def __init__(self, app: Optional[Flask] = None) -> None:
+        self.app = app
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app: Flask) -> 'None':
+        """This is used to set up session for your app object.
+        :param app: the Flask app object with proper configuration.
+        """
+        app.session_interface = self._get_interface(app)  # type: ignore
+
+    def _get_interface(self, app: Flask) -> SessionInterface:
+        config = app.config.copy()
+        config.setdefault('SESSION_REDIS', None)
+        config.setdefault('SESSION_LIFETIME', 2 * 60 * 60)
+        config.setdefault('SESSION_RENEW_COUNT', 5)
+        config.setdefault('SESSION_SIGNER_SALT', 'session')
+        config.setdefault('SESSION_KEY_PREFIX', 'session:')
+        config.setdefault('SESSION_HEADER_NAME', 'authorization')
+
+        session_interface = SessionInterface(
+            config['SESSION_LIFETIME'], config['SESSION_RENEW_COUNT'],
+            config['SESSION_REDIS'], config['SESSION_KEY_PREFIX'],
+            config['SESSION_SIGNER_SALT'], config['SESSION_HEADER_NAME'])
+
+        return session_interface
+
+
+def logout_user(uid: int, is_current: bool = True) -> None:
```

Comment:
`is_current` here is used to tell the function whether the current user is being logged out, or another user is being logged out. As an example, when an admin change a journalist password, then such user is logged out, but in that case we do not want to remove the `uid` from the admin user session, logging him out. It is probably not needed though, as self logout is handed differently (via `destroy()`). I'll double check and clean it up in case!

Reply:
Dropped the `is_current` parameter and the corresponding code paths as there were simpler way to achieve the same outcome.